### ISSUE TITLE
Manager scripts dont recieve certian events if the script is placed in a folder

### ISSUE THREAD

**Issue by HamishBrownPFR** [APSIMInitiative/ApsimX#8885]
### What happened?

I have noticed this for events broadcast from the plant model but there could be other cases.  In the example attached we have a manager doing a soil reset in a method attahced to the  "Sowing" event.  You can see if the manager is dirrectly below the zone it does the reset but if we move the manager into a folder it does not.

There is potential for alot of time to be wasted debugging manager scripts which could be overcome if these events connected properly.

![image](https://github.com/APSIMInitiative/ApsimX/assets/7295858/e419ccea-db54-4e9a-a6b3-9776bc42dc3c)
![image](https://github.com/APSIMInitiative/ApsimX/assets/7295858/33f033b2-9cba-4f3a-b023-6f44e2a3a9cd)
[BaseDevelopment (2).zip](https://github.com/APSIMInitiative/ApsimX/files/15172541/BaseDevelopment.2.zip)

### What Operating System are you on?

Windows

**Comment by hol353** [comment]
Yep this is a know problem - see #4423

**Comment by HamishBrownPFR** [comment]
Have we come up with a solution yet.  #4423 is quite broad regarding scoping.  Are there smaller parts we can pick of incrementally

**Comment by hol353** [comment]
No solution yet because the problem is tricky. @hut104 's solution is to not put manager scripts in a folder. I'd like a better fix though. Perhaps I try and find some time with the team to see if we can find a solution that isn't too hard to implement.

**Comment by HamishBrownPFR** [comment]
Not moving managers out of folders certainly solves the problem.   However, if you don't know to do that (or forgot) it can take a long time to work out why your manager is not working.  
We could change manager so Folder is not a valid parent (to stop the problem being replicated) and write a converter that moves all managers in folders in existing simulations up a level and deletes the folder.

**Comment by zur003** [comment]
I haven't really thought it through, but would there be any merit in having a second type (subclass?) of Folder for which the contents are always in scope?

**Comment by hol353** [comment]
@zur003 Yes, that is an option that I've thought about as well. I would rather find a scoping rule that works for all cases rather than introducing a 'special folder' that changes the rule for a particular edge case. We might end up going with the simple option though and do as Eric suggests.

**Comment by HamishBrownPFR** [comment]
@par456 is this still relevent?
